src/containers/my-cooperations/cooperation-notes/CooperationNotes.tsx

```diff
@@ -1,219 +1,170 @@
-  })
+  const createNote = (data: CreateOrUpdateNoteParams) => {
+    return CooperationNotesService.createNote(data, id)
+  }
 
-  const {
-    error: updateError,
-    loading: updateLoading,
-    fetchData: updateNote
-  } = useAxios({
-    service: updateNoteService,
-    fetchOnMount: false,
-    defaultResponse: null,
-    onResponseError,
-    onResponse: onUpdateResponse
+  const { mutate: addNewNote, isPending: createLoading } = useMutation({
+    mutationFn: createNote,
+    queryKey: ['notes'],
+    onError: handleErrorAlert,
+    onSuccess: () => onSuccessResponse('create')
   })
 
-  const handleDelete = async (id: string, isConfirmed: boolean) => {
-    if (isConfirmed) {
-      await deleteItem(id)
-      if (!error) await fetchData()
-    }
+  const deleteSelectedNote = (noteId: string) => {
+    return CooperationNotesService.deleteNote(id, noteId)
   }
 
+  const { mutate: deleteNote } = useMutation({
+    mutationFn: deleteSelectedNote,
+    queryKey: ['notes'],
+    onError: handleErrorAlert,
+    onSuccess: () => onSuccessResponse('delete')
+  })
+
   const onDeleteNote = (id: string) => {
     openDialog({
       message: 'cooperationsPage.modalMessages.confirmDeletionMessage',
-      sendConfirm: (isConfirmed: boolean) => void handleDelete(id, isConfirmed),
-      title: `cooperationsPage.modalMessages.confirmDeletionTitle`
+      title: `cooperationsPage.modalMessages.confirmDeletionTitle`,
+      sendConfirm: (isConfirmed: boolean) => {
+        if (isConfirmed) {
+          deleteNote(id)
+        }
+      }
     })
   }
 
-  const duplicateNote = useCallback(
-    (id?: string) => {
-      const note = notes.find((item) => item._id === id)
-      return createNoteService(note)
-    },
-    [notes, createNoteService]
-  )
-
-  const onDuplicateResponse = () => {
-    dispatch(
-      openAlert({
-        severity: snackbarVariants.success,
-        message: `cooperationsPage.modalMessages.successDuplication`
-      })
-    )
+  const updateSelectedNote = (params: {
+    noteId: string
+    data: CreateOrUpdateNoteParams
+  }) => {
+    return CooperationNotesService.updateNote(id, params.noteId, params.data)
   }
 
-  const { error: duplicationError, fetchData: duplicateItem } = useAxios({
-    service: duplicateNote,
-    fetchOnMount: false,
-    defaultResponse: null,
-    onResponseError,
-    onResponse: onDuplicateResponse
+  const { mutate: updateNote, isPending: updateNoteLoading } = useMutation({
+    mutationFn: updateSelectedNote,
+    queryKey: ['notes'],
+    onError: handleErrorAlert,
+    onSuccess: () => onSuccessResponse('update')
   })
 
-  const handleDuplicate = async (itemId: string) => {
-    await duplicateItem(itemId)
-    if (!duplicationError) await fetchData()
+  const duplicateNote = async (noteId: string) => {
+    const note = notes?.find((item) => item._id === noteId)
+    if (!note) {
+      return
+    }
+    return CooperationNotesService.createNote(note, id)
```

Comment:
Firstly, this function is not async, secondly - please, add spaces between `if` statements, it's unreadable now.

```suggestion
  const duplicateNote = (noteId: string) => {
    const note = notes?.find((item) => item._id === noteId)
    
    if (!note) {
      return
    }
    
    return CooperationNotesService.createNote(note, id)
```